application/client/src/app/service/recent.ts

```diff
@@ -62,7 +70,18 @@ export class Service extends Implementation {
     }
 
     public delete(uuids: string[]): Promise<void> {
-        return bridge.entries({ key: STORAGE_KEY }).delete(uuids);
+        return new Promise<void>((resolve, reject) => {
+            bridge
+                .entries({ key: STORAGE_KEY })
+                .delete(uuids)
+                .then(() => {
+                    this.updated.emit();
+                    resolve();
+                })
```

Comment:
I cannot select line 69, but looks like `updated` event should happen not only remove, but on adding too. It make sense to emit event also with it. See line 69